packages/mobile/src/home/WalletHome.tsx

```diff
@@ -90,12 +96,29 @@ export class WalletHome extends React.Component<Props> {
     this.props.refreshAllBalances()
   }
 
-  componentDidMount() {
+  componentDidMount = async () => {
     // TODO find a better home for this, its unrelated to wallet home
     this.props.initializeSentryUserContext()
     if (SHOW_TESTNET_BANNER) {
       this.showTestnetBanner()
     }
+    await this.tryImportContacts()
+  }
+
+  // TODO: cache recent recipients of txs to speed this up
+  tryImportContacts = async () => {
+    const { numberVerified, recipientCache } = this.props
+
+    // Only import contacts if number is verified and
+    // recip cache is empty so we haven't already
+    if (!numberVerified || recipientCache.length) {
+      return
+    }
+
+    const hasGivenContactPermission = await requestContactsPermission()
```

Comment:
I think you're looking for `checkContactsPermission` here

Reply:
Yup, didnt know that existed. Thanks!